### PR TITLE
Improve soft assert (#505 updated)

### PR DESCRIPTION
**Pull request by juherr**
Update #505

### CODE REVIEW

File: src/test/java/test/assertion/SoftAssertTest.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+import org.testng.asserts.IAssert;
+import org.testng.asserts.SoftAssert;
+
+import java.util.ArrayList;
+import java.util.Collection;
+
+public class SoftAssertTest {
+  @Test
+  public void testOnSucceedCalled() throws Exception {
+    final Collection<IAssert> succeed = new ArrayList<IAssert>();
```
Comment (cbeust) [inline src/test/java/test/assertion/SoftAssertTest.java]: Can be `ArrayList<>`.

File: src/test/java/test/assertion/SoftAssertTest.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+      }
+    };
+    sa.assertTrue(true);
+    sa.assertTrue(false);
+    Assert.assertEquals(succeed.size(), 1, succeed.toString());
+  }
+
+  @Test
+  public void testOnFailureCalled() throws Exception {
+    final Collection<IAssert> failured = new ArrayList<IAssert>();
```
Comment (cbeust) [inline src/test/java/test/assertion/SoftAssertTest.java]: `failured` not right. Maybe `failures`? Also, diamond operator.

File: src/test/java/test/assertion/SoftAssertTest.java [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    final SoftAssert sa = new SoftAssert() {
+      @Override
+      public void onAssertFailure(IAssert assertCommand, AssertionError ex) {
+        failured.add(assertCommand);
+      }
+    };
+    sa.assertTrue(true);
+    sa.assertTrue(false);
+    Assert.assertEquals(failured.size(), 1, failured.toString());
```
Comment (cbeust) [inline src/test/java/test/assertion/SoftAssertTest.java]: You could probably combine both tests into one and assert both that `failures.size() == 1` and that `successes.size() == 1`.

**Comment by cbeust** [comment]
LGTM with a couple of comments.

**Comment by juherr** [comment]
Done ;)

**Pull request merged**